LibOS/shim/src/sys/shim_mmap.c

```diff
@@ -79,70 +94,38 @@ void * shim_do_mmap (void * addr, size_t length, int prot, int flags, int fd,
         }
     }
 
-    void * mapped = ALIGN_DOWN((void *) addr);
-    void * mapped_end = ALIGN_UP((void *) addr + length);
-
-    addr = mapped;
-    length = mapped_end - mapped;
-
-    if (flags & MAP_ANONYMOUS) {
+    if (!hdl) {
         addr = (void *) DkVirtualMemoryAlloc(addr, length, pal_alloc_type,
                                              PAL_PROT(prot, 0));
 
         if (!addr) {
-            ret = (PAL_NATIVE_ERRNO == PAL_ERROR_DENIED) ? -EPERM : -PAL_ERRNO;
-            goto free_reserved;
+            if (PAL_NATIVE_ERRNO == PAL_ERROR_DENIED)
```

Comment:
Note for the future: `PAL_NATIVE_ERRNO` should be changed to a function-like macro, otherwise such comparisons look terrible: it looks like you were comparing two constants here.